backend-test/Controllers/CourseControllerTests.hs

```diff
@@ -147,6 +147,88 @@ runDeptsTest label courses expected =
+                , coursesDistribution = Nothing
+                , coursesPrereqString = Just "(  MAT135H1,  MAT136H1)/  MAT137Y1/  MAT157Y1/  (  MATA30H3,  MATA36H3)/  (  MATA31H3,  MATA37H3)/  (  MAT135H5,  MAT136H5)/  MAT137Y5/  MAT157Y5/  ( MAT137H5,  MAT139H5)/  ( MAT157H5,  MAT159H5)"
+                , coursesCoreqs = Just "(  CSC108H1/  equivalent programming experience)/  CSC110Y1/  CSC148H1 *Note: the corequisite may be completed either concurrently or in advance."
+                , coursesVideoUrls = []
+                }
+            sta238 = Courses
+                { coursesCode = "STA238H1"
+                , coursesTitle = Just "Probability, Statistics and Data Analysis II"
+                , coursesDescription = Just "An introduction to statistical inference and practice. Statistical models and parameters, estimators of parameters and their statistical properties, methods of estimation, confidence intervals, hypothesis testing, likelihood function, the linear model. Use of statistical computation for data analysis and simulation."
+                , coursesPrereqs = Just "STA237H1/  STA247H1/  STA257H1/  STAB52H3/  STA256H5"
+                , coursesExclusions = Just "ECO220Y1/  ECO227Y1/  GGR270H1/  PSY201H1/  SOC300H1/  SOC202H1/  SOC252H1/  STA220H1/  STA221H1/  STA255H1/  STA248H1/  STA261H1/  STA288H1/  EEB225H1/  STAB22H3/  STAB27H3/  STAB57H3/  STA220H5/  STA221H5/  STA258H5/  STA260H5/  ECO220Y5/  ECO227Y5"
+                , coursesBreadth = Nothing
+                , coursesDistribution = Nothing
+                , coursesPrereqString = Just "STA237H1/  STA247H1/  STA257H1/  STAB52H3/  STA256H5"
+                , coursesCoreqs = Just "CSC108H1/  CSC110Y1/  CSC148H1 *Note: the corequisite may be completed either concurrently or in advance."
+                , coursesVideoUrls = [T.pack "[https://example.com/video1",T.pack "https://example.com/video2]"]
+                }
+            csc108 = Courses
+                { coursesCode = "CSC108H1"
+                , coursesTitle = Just "Introduction to Computer Programming"
+                , coursesDescription = Just "Programming in a language such as Python. Elementary data types, lists, maps. Program structure: control flow, functions, classes, objects, methods. Algorithms and problem solving. Searching, sorting, and complexity. Unit testing. Floating-point numbers and numerical computation. No prior programming experience required."
+                , coursesPrereqs = Nothing
+                , coursesExclusions = Just "CSC110Y1,  CSC111H1,  CSC120H1,  CSC121H1,  CSC148H1,  CSC108H5,  CSC148H5,  CSCA08H3,  CSCA20H3,  CSCA48H3"
+                , coursesBreadth = Nothing
+                , coursesDistribution = Nothing
+                , coursesPrereqString = Nothing
+                , coursesCoreqs = Nothing
+                , coursesVideoUrls = []
+                }
+
+        runDb $ do
+            clearDatabase
+            when (state == 1) $ do 
```

Comment:
So overall it's a bit clumsy to use `state` just for this purpose. (Aside: `Bool` would have been a better datatype than `Int` for your use of `state`.)

Instead, make the type of `courseInfoTestCases :: [(String, [Courses], T.Text, String)]`, where now the second value in the tuple is a list of courses to insert. You can move the definitions of `sta237`, `sta238`, and `csc108` into a `where` block for that definition above.